File: backend/app/schemas/mannschaft.py
# Copyright (c) 2026 Noah, Luca, Sheila, Lando. All rights reserved.

from typing import Optional

from pydantic import BaseModel, ConfigDict


class MannschaftCreate(BaseModel):
    name: str
    rang: int
    spielklasse: Optional[str] = None
    fuehrer_id: Optional[int] = None


class MannschaftUpdate(BaseModel):
    name: Optional[str] = None
    rang: Optional[int] = None
    spielklasse: Optional[str] = None
    fuehrer_id: Optional[int] = None


class MitgliedCreate(BaseModel):
    spieler_id: int
    standardposition: Optional[int] = None
    ist_stammspieler: bool = True
    meldenummer: Optional[str] = None


class MitgliedRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    spieler_id: int
    standardposition: Optional[int]
    ist_stammspieler: bool
    meldenummer: Optional[str] = None


class MannschaftRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    rang: int
    spielklasse: Optional[str]
    fuehrer_id: Optional[int]

File: frontend/lib/models/mannschaft.dart
// Copyright (c) 2026 Noah, Luca, Sheila, Lando. All rights reserved.

class Mannschaft {
  final int id;
  final String name;
  final int rang;
  final String? spielklasse;
  final int? fuehrerId;

  Mannschaft({
    required this.id,
    required this.name,
    required this.rang,
    this.spielklasse,
    this.fuehrerId,
  });

  factory Mannschaft.fromJson(Map<String, dynamic> j) => Mannschaft(
        id: j['id'] as int,
        name: j['name'] as String,
        rang: j['rang'] as int,
        spielklasse: j['spielklasse'] as String?,
        fuehrerId: j['fuehrer_id'] as int?,
      );
}
